File: DSAL/Lab1/linear_search.c
//program to find an integer element using linear search

#include <stdio.h>
#include <stdbool.h>
int Lsearch(int arr[], int n, int k)
{
    int check = -1;
    int i = 0;
    for(i=0; i<n; i++)
    {
        if(arr[i] == k)
        {
            check = i;
            break;
        }
    }
    return check+1;
}
int main()
{
    int i, n, k;
    printf("Enter the size of array: ");
    scanf("%d", &n);
    int arr[n];
    printf("\nEnter the elements of the array: ");
    for(i=0; i<n; i++)
        scanf("%d", &arr[i]);
    printf("\nEnter the element to be searched: ");
    scanf("%d", &k);
    int check = Lsearch(arr, n, k);
    (check == -1) ? printf("%d is not present in the array", k) : printf("%d is present in the array at position %d", k, check);
    return 0;
}

File: OOPL/lab2/linearSearch.java
//program to search an element using linear search in java

import java.util.Scanner;
class linearSearch {
	public static void main (String[] args) {
		Scanner sc = new Scanner(System.in);

		System.out.println("Enter the size of the array: ");
		int size = sc.nextInt();

		int arr[] = new int[size];

		System.out.println("Enter the elements into the array: ");
		for(int i=0; i<size; i++)
			arr[i] = sc.nextInt();

		System.out.println("Enter the elements to be searched: ");
		int k = sc.nextInt();

		int i;
		for(i=0; i<size; i++)
			if(arr[i] == k)
			{
				System.out.println(arr[i]+" is at index "+(i+1));
				break;
			}

			if(i == size)
				System.out.println("Element not found");

	}
}
